frontend/src/utils/requirelogin.tsx

```diff
@@ -71,3 +73,23 @@ export function requireVerifiedEmail<T>(
     );
   });
 }
+
+// following functions should only be used in roadmap routes, since a roadmap role is assumed
+export const requireRole = <T,>(
+  Component: ComponentType<T>,
+  required: RoleType[],
+): FC<T> => (props) => {
+  const role = useSelector(userRoleSelector, shallowEqual);
+  if (!role || !required.includes(role))
+    return <Redirect to={paths.notFound} />;
+  return <Component {...props} />;
+};
+
+export const requirePermission = <T,>(
+  Component: ComponentType<T>,
+  required: Permission,
+): FC<T> => (props) => {
+  const role = useSelector(userRoleSelector, shallowEqual);
+  if (!hasPermission(role, required)) return <Redirect to={paths.notFound} />;
+  return <Component {...props} />;
+};
```

Comment:
I'd rename these functions in a fashion that the comment wouldn't be needed. But I'll leave this up to you to decide since the comment does the same job anyways, I just prefer minimal comments with more explanatory naming.